prometheus/collect_topic_partition_offsets.go

```diff
@@ -101,5 +101,56 @@ func (e *Exporter) collectTopicPartitionOffsets(ctx context.Context, ch chan<- p
 		}
 	}
 
+	// Highest Timestamp Offsets
+	// NB: this requires Kafka Brokers 3.0+ (see https://issues.apache.org/jira/browse/KAFKA-12541)
+	// In older versions this is returning the timestamp of the low watermarks (earliest offset)
+	maxTimestampOffsets, err := e.minionSvc.ListOffsetsCached(ctx, -3)
+	if err != nil {
```

Comment:
are we expecting an error for brokers with version < 3.0.0 ?

Reply:
Not an error but incorrect results:
"""In older versions this is returning the timestamp of the low watermarks (earliest offset)"""

Reply:
thanks for the clarification.